ibm_cloud_sdk_core/jwt_token_manager.py

```diff
@@ -80,6 +92,42 @@ def set_disable_ssl_verification(self, status: bool = False):
         """
         self.disable_ssl_verification = status
 
+    def paced_request_token(self):
+        """
+        Paces requests to request_token.
+
+        This method pseudo-serializes requests for an access_token
+        when the current token is expired (or has never been fetched).
+        The first caller into this method records its `request_time` and
+        then issues the token request. Subsequent callers will check the
+        `request_time` to see if a request is active (has been issued within
+        the past 60 seconds), and if so will sleep for a short time interval
+        (currently 1 second) before checking again. The check for an active
+        request and update of `request_time` are serailized by the `lock`
+        variable so that only one caller can become the active requester
+        with a 60 second interval.
+
+        Threads that sleep waiting for the active request to complete will
+        eventually find a newly valid token and return, or 60 seconds will
+        elapse and a new thread will assume the role of the active request.
+        """
+        while self._is_token_expired():
+            current_time = self._get_current_time()
+
+            with self.lock:
+                request_active = self.request_time > (current_time - 60)
+                if not request_active:
+                    self.request_time = current_time
+
+            if not request_active:
+                token_response = self.request_token()
+                self._save_token_info(token_response)
+                self.request_time = 0
+                return
+
+            time.sleep(1)  # Sleep for 1 second before checking token again
```

Comment:
I think this might be a little on the long side, I can do a token exchange in 220 ms (with 18 ms ping time from my local laptop and I've seen ping times < 1 ms when running in Cloud). I think 0.1 might be closer to the mark if we don't want to bottleneck for any longer than necessary.

Reply:
This is actually a big improvement from the previous version of the code, where we might have slept for up to 60 seconds before checking again for a valid token 😄 .

I'm not really that concerned about a small bottleneck here, since it should be a relatively rare occurrence that we ever execute this code.  But I'm willing to cut it half, to 0.5s, to make it just a little more snappy.